bin/metrics-vsphere.rb

```diff
@@ -0,0 +1,353 @@
+            "virtualDisk.writeLoadMetric",
+            "virtualDisk.readIOSize",
+            "virtualDisk.writeIOSize",
+            "virtualDisk.smallSeeks",
+            "virtualDisk.mediumSeeks",
+            "virtualDisk.largeSeeks",
+            "virtualDisk.readLatencyUS",
+            "virtualDisk.writeLatencyUS",
+            "virtualDisk.vFlashCacheIops",
+            "virtualDisk.vFlashCacheLatency",
+            "virtualDisk.vFlashCacheThroughput",
+            "datastore.numberReadAveraged",
+            "datastore.numberWriteAveraged",
+            "datastore.read",
+            "datastore.write",
+            "datastore.totalReadLatency",
+            "datastore.totalWriteLatency",
+            "datastore.sizeNormalizedDatastoreLatency",
+            "datastore.datastoreIops",
+            "datastore.datastoreReadBytes",
+            "datastore.datastoreWriteBytes",
+            "datastore.datastoreReadIops",
+            "datastore.datastoreWriteIops",
+            "datastore.datastoreNormalReadLatency",
+            "datastore.datastoreNormalWriteLatency",
+            "datastore.datastoreReadOIO",
+            "datastore.datastoreWriteOIO",
+            "datastore.datastoreMaxQueueDepth",
+            "datastore.datastoreReadLoadMetric",
+            "datastore.datastoreWriteLoadMetric",
+            "datastore.maxTotalLatency",
+            "datastore.siocActiveTimePercentage",
+            "datastore.datastoreVMObservedLatency",
+            "power.power",
+            "power.powerCap",
+            "power.energy",
+            "hbr.hbrNumVms",
+            "hbr.hbrNetRx",
+            "hbr.hbrNetTx",
+            "vflashModule.numActiveVMDKs",
+            "vsanDomObj.readIops",
+            "vsanDomObj.readThroughput",
+            "vsanDomObj.readAvgLatency",
+            "vsanDomObj.readMaxLatency",
+            "vsanDomObj.readCacheHitRate",
+            "vsanDomObj.readCongestion",
+            "vsanDomObj.writeIops",
+            "vsanDomObj.writeThroughput",
+            "vsanDomObj.writeAvgLatency",
+            "vsanDomObj.writeMaxLatency",
+            "vsanDomObj.writeCongestion",
+            "vsanDomObj.recoveryWriteIops",
+            "vsanDomObj.recoveryWriteThroughput",
+            "vsanDomObj.recoveryWriteAvgLatency",
+            "vsanDomObj.recoveryWriteMaxLatency",
+            "vsanDomObj.recoveryWriteCongestion",
+            "gpu.utilization",
+            "gpu.mem.used",
+            "gpu.mem.usage",
+            "gpu.temperature",
+            "pmem.available.reservation"]
+
+
+  def vim
+    @vim ||= RbVmomi::VIM.connect(host: config[:host], user: config[:user], password: config[:password], insecure: config[:insecure])
+  end
+
+  def unit_for_metric(metric)
+    metric_array = metric.split('.')
+    group_name = metric_array.shift
+    counter_name = metric_array.join('.')
+
+    @json ||= JSON.parse(File.read(File.join(File.dirname(__FILE__), '..', 'lib', 'vcenter.json')))
```

Comment:
Rather than reading in the file from json and then turn it into a ruby hash. Why not just create a library that sets a constant hash similar to what we do for setting the [version](https://github.com/sensu-plugins/sensu-plugins-vsphere/blob/master/lib/sensu-plugins-vsphere/version.rb)?

I am not saying this is wrong I have just not really seen anyone really do this. Is there some benefit such as automatically generating that json blob via an api?